### ISSUE TITLE
RStudio 2023.12.0+369 fails to startup on projects?

### ISSUE THREAD

**Issue by eyayaw** [rstudio/rstudio#14071]
Rstudio normally starts in a non-rstudio project environment. However, it does not start up with rstudio projects, and takes a long time to throw the below error. 

As the error log shows, the error seems to be that the `rmarkdown` package can't be found. However, I have it installed. 

```
R version 4.3.2 (2023-10-31) -- "Eye Holes"
Copyright (C) 2023 The R Foundation for Statistical Computing
Platform: aarch64-apple-darwin23.0.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.
```
```r
R> "rmarkdown" %in% installed.packages()[, "Package"]
[1] TRUE

```

<details>
<summary>R Session Startup Failure Report</summary>

```
## R Session Startup Failure Report

### RStudio Version

RStudio 2023.12.0+369 "Ocean Storm " (33206f75, 2023-12-14) for macOS

Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.12.0+369 Chrome/116.0.5845.190 Electron/26.2.4 Safari/537.36

### Error message

[No error available]

### Process Output

The R session exited with code 1. 

Error output:

```
[No errors emitted]
```

Standard output:

```
[No output emitted]
```

### Logs

*/Users/eyayaw/.local/share/rstudio/log/rsession-eyayaw.log*

```
2023-12-20T13:08:05.944665Z [rsession-eyayaw] ERROR r error 4 (Error in loadNamespace(x) : there is no package called ‘rmarkdown’
) [errormsg: Error in loadNamespace(x) : there is no package called ‘rmarkdown’
]; OCCURRED AT rstudio::core::Error rstudio::r::exec::(anonymous namespace)::evaluateExpressionsUnsafe(SEXP, SEXP, SEXP *, sexp::Protect *, rstudio::r::exec::(anonymous namespace)::EvalType) src/cpp/r/RExec.cpp:231; LOGGED FROM: bool rstudio::session::module_context::isBookdownProject() src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp:1890

```
                
```
</details>

![SCR-20231220-mzsx](https://github.com/rstudio/rstudio/assets/49317723/e7992c76-4840-4315-a6f1-42a95c5a4030)

**Comment by ronblum** [comment]
@eyayaw Thank you for raising the issue! A few questions:

1. Do you mean by double-clicking on a `.proj` file? If so, what happens if you open from a `.R` and `.Rmd` file?
2. If you open it not in a project, are you able to then open the project successfully?
3. What version of MacOS are you using?

**Comment by eyayaw** [comment]
@ronblum thank you for the reply. 

I could open RStudio and navigate to the project folder. However, clicking on the .rproj file does not work. I noticed that it works in some project files. I thought that if `renv` has something to do with it, I removed `.Rprofile`, `renv.lock`, and `renv/` and tried again, but it did not work.

**Comment by eyayaw** [comment]
Reverted to https://dailies.rstudio.com/version/2023.09.0+463.pro11/, and it works. Looks like it has to do with the latest version. 

It is **macOS Sonoma 14.2.1**.

**Comment by kevinushey** [comment]
Based on the error, it looks like RStudio is failing to start up because R is trying to load the `rmarkdown` package on startup when attempting to determine if the project is a 'bookdown' project, as per here:

https://github.com/rstudio/rstudio/blob/6f7af1eab02568be3de327190a96fc5dba2dc247/src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp#L1880-L1892

https://github.com/rstudio/rstudio/blob/6f7af1eab02568be3de327190a96fc5dba2dc247/src/cpp/session/modules/SessionRMarkdown.R#L497-L508

I think this is worth taking a look at for the patch release. We should at least:

1. Properly verify whether `rmarkdown` is installed before using it, or
2. Avoid using `rmarkdown` if at all possible when inferring if a project is a bookdown project.

**Comment by kevinushey** [comment]
Here's what I did to try and reproduce:

1. File -> New Project...; New Directory -> Book project using bookdown -> example
2. After the bookdown project is open, remove the `rmarkdown` package,
3. Close the RStudio IDE, and then re-open it.

In my case, I don't see a failure to start, though -- instead, I see errors in the Console on startup.

```
Error in loadNamespace(x) : there is no package called ‘rmarkdown’
Error in loadNamespace(x) : there is no package called ‘rmarkdown’
Error in loadNamespace(x) : there is no package called ‘rmarkdown’
Error in loadNamespace(x) : there is no package called ‘rmarkdown’
Error in loadNamespace(x) : there is no package called ‘rmarkdown’
```

In other words, it's possible that the `rmarkdown` error in the logs is a red herring, but this is still worth cleaning up.

**Comment by gpertea** [comment]
This also happens to me with this 369 version on a few machines (Ubuntu 20 and Ubuntu 22) -- I cannot open recent projects that had no such problems before the update. 
The desktop version just hangs on a blank loading screen, while rsession processes start running in the background, taking gigabytes of RAM -- but the UI never changes from the blank page.  No error messages I can see (at first I did have some dbus warnings that forced me to install some dbus services, hoping that would fix it, but it did not help)

Reverting to 2023.09.1-494 solves the problem, I can load those same projects quickly.

**Comment by kevinushey** [comment]
@gpertea: I'd start by resetting your RStudio state, following the instructions in https://support.posit.co/hc/en-us/articles/200534577-Resetting-RStudio-Desktop-s-State. You might also want to try removing the project's `.Rproj.user` folder, or moving it out of the way temporarily.

**Comment by gpertea** [comment]
@kevinushey : I remember trying that, did not help.  (if it did, that would've meant there was some undocumented incompatibility between the new version and the old RStudio or project settings, which still sounds like a bug in the new version, not a new feature..).

I am puzzled by the fact that the same projects load fine & quickly as soon as I switch back to 2023.09.1-494 (with or without clean-up/resetting Rstudio and project settings/cache etc.). 

I'm using the .deb files for the standard (non-Pro) version.

**Comment by kevinushey** [comment]
> I am puzzled by the fact that the same projects load fine & quickly as soon as I switch back to 2023.09.1-494 (with or without clean-up/resetting Rstudio and project settings/cache etc.).

Is there anything that makes these projects unique versus other projects you're working on? Any chance you can figure out how to get RStudio into this state, so that I might try and reproduce locally?

**Comment by gpertea** [comment]
Just tried 2023.12.1-402 (server) on Ubuntu 20 and I get similar blank screen/stuck behavior only on a single project that overwise opens OK with 2023.09.1-494

If it matters, I access this RStudio server through a ssh tunnel, so the url has an unusual port number, even though the server itself runs on 8787.  However this configuration works OK for me with any other projects on the same server/tunnel, except this odd one. 

It might be related to the fact that this project has many symlinked files and directories in its main directory, plus many extraneous files.  

However, simply cloning the github repository for this directory again in a fresh new local directory  makes the project load OK ! 

I do not think it's related to the `.Rproj.user` folder not being present in the fresh clone because I tried renaming that folder in the original project directory but that did not help. 

The freshly cloned directory has a much lower number of files and no symlinks. Not sure why this matters for the 2023.12.1-402 (while it did not for 2023.09.1-494).

P.S. the other projects that load OK also have symlinked files/directories so just the presence of symlinks is not the issue, but without a proper startup log It's hard to debug this with the old project directory, I suppose. 

However I think right now **my issue can be considered resolved**, I'll just work in the fresh new cloned directory, creating symlinks as needed, and I can submit another issue if the problem returns.

**Comment by ronblum** [comment]
@gpertea Oops I forgot to add, good to know that you got it working, and please let us know if anything again goes awry!
